File: Liveramp2.c
#include<stdio.h>
#include<stdlib.h>



int find_left_of_arr( int * arr, int length)
{
	int i;
	for ( i=1; i< length; i++)
	{
		if ( arr[i] <arr[i-1])
		{
			break;
	
		}
	}
	return i-1;
}

int find_right_of_arr( int * arr, int length)
{
	for ( int i=length-2;i>=0; i--)
	{
		if ( arr[i] > arr[i+1])
		{
			return i+1;
	
		}
	}
	return 0;	

}


int shrinkleft ( int * arr, int min, int findleft)
{

	int i;
	for (i = findleft; i>=0; i--)
	{
		if ( arr[i] <= arr[min])
		{
			break;
		}
		
	}
	
	return i+1;

}

int shrinkright ( int * arr, int max, int findright, int length)
{

	int i;
	for (i= findright; i< length; i++)
	{
		if ( arr[i] >= arr[max] )
		{
			break;
		}
		
	}
	return i-1;
		
}

int findsequence( int * arr, int length)
{


	int findleft;
	findleft = find_left_of_arr(arr,length);

	if ( findleft >= length-1 )
	{
		return 0;
	}

	int findright;
	findright = find_right_of_arr(arr,length);
	
	int min,max;
	
	min = findright;
	max = findleft;

	for ( int i = findleft +1; i< findright; i++)
	{
		if ( arr[i] < arr[min])
		{
			min = i;
		}
		if ( arr[i] > arr[max])
		{
			max = i;
		}
	}



	int finalleftindex;
	int finalrightindex;

	finalleftindex = shrinkleft(arr,min,findleft);
	finalrightindex= shrinkright(arr,max,findright,length);
	int max2 = arr[finalleftindex]>arr[max]?finalleftindex:max;
	int min2 = arr[finalrightindex]<arr[min]?finalrightindex:min;

	while(min2!=min || max2!=max){
		min2 = min;max2=max;
		finalleftindex = shrinkleft(arr,min,findleft);
		finalrightindex= shrinkright(arr,max,findright,length);
		max2 = arr[finalleftindex]>arr[max]?finalleftindex:max;
		min2 = arr[finalrightindex]<arr[min]?finalrightindex:min;
		
	}
	return (finalrightindex - finalleftindex);

}






int main()
{
		int ret;
		int array[8] = {1,2,6,11,6,5,18,19};

		int length = (sizeof(array)/sizeof(array[0]));
		ret = findsequence(array,length);
		printf("%d",ret);

		return 0 ;
}
	

File: Liveramp2.py
# coding=utf-8
def find_left_of_arr(arr, length):
    for i in range(1,length):
        if arr[i]<arr[i-1]:
            break
    return i-1

def find_right_of_arr(arr,length):
    for i in range(length-2,-1,-1):
        if arr[i] > arr[i+1]:
            return i+1
    return 0

def shrinkleft(arr,min,max,findleft):
    for i in range(findleft,-1,-1):
        if arr[i]<=arr[min]:
            break
    max2 = i+1 if arr[i+1]>arr[max] else max
    return i+1, max2

def shrinkright(arr,min,max,findright,length):
    for i in range(findright,length):
        if arr[i]>=arr[max]:
            break
    min2 = i-1 if arr[i-1]<arr[min] else min
    return i-1, min2

def findsequence(arr,length):
    findleft = find_left_of_arr(arr,length)
    print findleft
    if findleft>=length-2:
        return 0
    findright = find_right_of_arr(arr,length)
    print findright
    min,max = findright,findleft
    for i in range(findleft+1,findright):
        if arr[i]<arr[min]:
            min = i
        if arr[i]>arr[max]:
            max = i


    finalleftindex,max2 = shrinkleft(arr,min,max,findleft)
    finalrightindex,min2 = shrinkright(arr,min,max,findright,length)
    '''while(max != max2 or min != min2):
        max, min = max2, min2
        finalleftindex, max2 = shrinkleft(arr, min, max, findleft)
        finalrightindex, min2 = shrinkright(arr, min, max, findright, length)'''

    return (finalrightindex-finalleftindex)

array = [1,2,6,0,0,18,19]
length = len(array)
print(findsequence(array,length))